src/module/item/abstract-effect/document.ts

```diff
@@ -12,6 +14,28 @@ export abstract class AbstractEffectPF2e extends ItemPF2e {
     abstract increase(): Promise<void>;
     abstract decrease(): Promise<void>;
 
+    /** Get the actor that created this effect */
+    get origin(): { actor: ActorPF2e | null; item: Embedded<ItemPF2e> | null } {
+        const [actorOrToken, item] = this.isOfType("affliction", "effect")
```

Comment:
given that condition overrides this getter, I don't think you have to check the type right?

Reply:
Only afflictions and effects have a `context` in their system data

Reply:
Typescript thing it is!